packages/core/saved-objects/core-saved-objects-migration-server-internal/src/kibana_migrator_utils.ts

```diff
@@ -0,0 +1,136 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import type { ElasticsearchClient } from '@kbn/core-elasticsearch-server';
+import { Logger } from '@kbn/logging';
+import type { IndexMap } from './core';
+import {
+  type TypeIndexMap,
+  TypeStatus,
+  TypeStatusDetails,
+  MAIN_SAVED_OBJECT_INDEX,
+} from './kibana_migrator_constants';
+
+// even though this utility class is present in @kbn/kibana-utils-plugin, we can't easily import it from Core
+// aka. one does not simply reuse code
+export class Defer<T> {
+  public resolve!: (data: T) => void;
+  public reject!: (error: any) => void;
+  public promise: Promise<any> = new Promise<any>((resolve, reject) => {
+    (this as any).resolve = resolve;
+    (this as any).reject = reject;
+  });
+}
+
+export const defer = () => new Defer<void>();
+
+export function createMultiPromiseDefer(indices: string[]): Record<string, Defer<void>> {
+  const defers: Array<Defer<void>> = new Array(indices.length).fill(true).map(defer);
+  const all = Promise.all(defers.map(({ promise }) => promise));
+  return indices.reduce<Record<string, Defer<any>>>((acc, indexName, i) => {
+    const { resolve, reject } = defers[i];
+    acc[indexName] = { resolve, reject, promise: all };
+    return acc;
+  }, {});
+}
+
+export async function checkTypeIndexDistribution(
+  client: ElasticsearchClient,
+  desiredIndexMap: TypeIndexMap,
+  defaultIndexTypeMap: TypeIndexMap
+): Promise<Record<string, TypeStatusDetails>> {
+  const mapping = await client.indices.getMapping({
+    index: MAIN_SAVED_OBJECT_INDEX,
+  });
+  const meta = Object.values(mapping)?.[0].mappings._meta;
+  const currentIndexMap: TypeIndexMap = meta?.typeIndexMap ?? defaultIndexTypeMap;
```

Comment:
When migrating from `< 8.8.0` the `.kibana.mappings._meta` will NOT have the `typeIndexMap` information. In this case, we assume that indices have the standard `type => index` distribution that is valid up until 8.8.0:
```
- 'task' saved objects are stored in '.kibana_task_manager' index
- everything else goes to '.kibana' index